### PR TITLE
Thresholding work for ROI heatmap experiment

### PR DESCRIPTION
**Pull request by EBruda**

### CODE REVIEW

File: fastmri/pl_modules/unet_module_heatmap.py [hunk 1]
```diff
@@ -96,52 +96,91 @@ def forward(self, image):
...
-        # This is from benchmark
         self.train_outputs[batch.fname[0]].append((batch.slice_num, output * std + mean))
         
-        # Calculate loss
-        loss = F.l1_loss(output, batch.target)
+        loss = self.weighted_loss_function(output, batch.target, roi_mask_tensor, roi_weight=2.0)
         self.log("loss", loss.detach())
         
-        # Return just the loss
+        if batch_idx % 50 == 0:
```
Comment (EBruda) [inline fastmri/pl_modules/unet_module_heatmap.py]: We would need to save the ROI mask for all inputs, not just every 50 batches

File: fastmri/pl_modules/unet_module_heatmap.py [hunk 2]
```diff
@@ -96,52 +96,91 @@ def forward(self, image):
...
         slice_num = int(batch.slice_num)
 
-        self._original_image(recon,  fname, slice_num, "val", "recon")
+        self._original_image(recon, fname, slice_num, "val", "recon")
         self._original_image(target, fname, slice_num, "val", "target")
+        self._original_image(input_image, fname, slice_num, "val", "input")
         self._heatmap(error_map, fname, slice_num, split="val")
+        
+        # Save the ROI mask
+        self._save_roi_mask(roi_mask, fname, slice_num, "val")
```
Comment (EBruda) [inline fastmri/pl_modules/unet_module_heatmap.py:159]: This is good! But we also need to save the ROI mask for test data as well, since if we can't report the reconstruction accuracy in those ROIs (which is the main research question)
